pkgs/shells/nushell/default.nix

```diff
@@ -39,6 +39,13 @@ rustPlatform.buildRustPackage rec {
     export HOME=$TMPDIR
   '';
 
+  checkPhase = ''
+    runHook preCheck
+    echo "Running cargo test"
+    cargo test
+    runHook postCheck
+  '';
```

Comment:
It should be sufficient to use `checkType = "debug";`.

Reply:
Indeed I tried that, as I've seen it in other derivations using `buildRustPackage`, but the `checkPhase` did still fail.

Reply:
Can you tell me *what* went wrong?

Reply:
The same as with the release build, here is a pastebin, when I use `checkPhase = "debug";` instead in the derivation:
https://pastebin.com/yRjpS2Fz

I tried a few different things, and the only thing that seemed to help, was, what is in currently in the PR (but as I'm still relatively new to nix/nixpkgs, I'm not 100% sure what I'm doing).
Maybe it has to do something with other things that are executed in the original `checkPhase`?
Errors like `Can't run test No such file or directory (os error 2)` seem to indicate that at least...

Reply:
>  Maybe it has to do something with other things that are executed in the original checkPhase?
Errors like Can't run test No such file or directory (os error 2) seem to indicate that at least...

Hmm, those tests have probably certain expectations about the structure of the `target/` directory. Since we internally use `cargo --target`, the target-directory actually looks like `target/<arch>/debug/*`, the test probably fails since it expects `target/debug`.

Both options are fine according to cargo docs, but a lot of packages don't care for the first option.

Unless this can be easily fixed (with an upstream patch), this change is fine from my PoV.

Reply:
I've created an issue (https://github.com/nushell/nushell/issues/2021). Until this is fixed, I would also suggest merging this to unbreak master.

Reply:
Yes, LGTM. Sorry, forgot about it.